sql: enable tenant testing for more tests

shubhamdhama opened the pull request:
To achieve this, this PR reduces the usage of `createTestServerParams` by replacing it with `createTestServerParamsAllowTenants`. The changes are mostly straightforward, including replacing hardcoded `keys.SystemSQLCodec` with `server.Codec()`, and using `server.StorageLayer()`/`server.SystemLayer()` where the test is specific to the system tenant.

Informs: #140446
Epic: CRDB-38970
Release note: None

shubhamdhama committed:
sql: enable tenant testing for more tests

To achieve this, this PR reduces the usage of `createTestServerParams`
by replacing it with `createTestServerParamsAllowTenants`. The changes
are mostly straightforward, including replacing hardcoded
`keys.SystemSQLCodec` with `server.Codec()`, and using
`server.StorageLayer()`/`server.SystemLayer()` where the test is
specific to the system tenant.

Informs: #140446
Epic: CRDB-38970
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/140447)
<!-- Reviewable:end -->

shubhamdhama commented:
There are still a lot of these usages left, but to keep the PR relatively easier for review I'll put it in next one.

shubhamdhama commented on pkg/sql/conn_executor_test.go:
```diff
@@ -119,7 +119,7 @@ func TestSessionFinishRollsBackTxn(t *testing.T) {
 	aborter := NewTxnAborter()
 	defer aborter.Close(t)
-	params, _ := createTestServerParams()
+	params, _ := createTestServerParamsAllowTenants()
```
This one is a problem with shared-process mode but not with external.

shubhamdhama replied:
Took care of this in: https://github.com/cockroachdb/cockroach/pull/140662

shubhamdhama replied:
Finally I decided to solve it in this PR itself since there is an existing handy API to give tenant specific connection string `s.ApplicationLayer().PGUrl()`

shubhamdhama committed:
testutils: fix ambiguous comment for `ApplicationLayer()`

The previous comment implied that `ApplicationLayer()` could randomly
return either the system tenant's or a secondary tenant's interface.
This change clarifies that behavior.

cthumuluru-crdb commented:
bors r+

rafiss reviewed (approved):
lgtm! thank you for improving these tests!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/140447#-:-OInSFOGBhVntLO-wZcS:b-8g9naq)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @cthumuluru-crdb)



<!-- Sent from Reviewable.io -->

shubhamdhama commented:
bors r=rafiss

shubhamdhama commented:
Thanks for the review, Rafi.

cthumuluru-crdb commented:
LGTM. Thanks, @shubhamdhama!

cthumuluru-crdb reviewed (commented):
LGTM!
